go/libraries/doltcore/sqle/writer/prolly_table_writer.go

```diff
@@ -78,15 +79,30 @@ func getSecondaryProllyIndexWriters(ctx context.Context, t *doltdb.Table, sqlSch
 		// mapping from secondary index key to primary key
 		pkMap := makeIndexToIndexMapping(def.Schema().GetPKCols(), sch.GetPKCols())
 
-		writers[defName] = prollySecondaryIndexWriter{
-			name:    defName,
-			mut:     idxMap.Mutate(),
-			unique:  def.IsUnique(),
-			idxCols: def.Count(),
-			keyMap:  keyMap,
-			keyBld:  val.NewTupleBuilder(keyDesc),
-			pkMap:   pkMap,
-			pkBld:   val.NewTupleBuilder(pkDesc),
+		prefixLengths := def.GetPrefixLengths()
```

Comment:
I think we always pass the prefix lengths and just branch depending on whether there is or isn't a prefix